### ISSUE TITLE
`flatpak run --command=bash --devel app` prints some bash errors 

### ISSUE THREAD

**Issue by matzipan** [flatpak/flatpak#381]
Whether I do `flatpak run --command=bash --devel org.gnome.Epiphany` or `flatpak run --command=bash --devel org.gnome.Builder`, bash prints the following error: 

```
bash: __bp_precmd_invoke_cmd: command not found
bash: __bp_interactive_mode: command not found
```

I am on Ubuntu 16.04, with flatpak 0.6.13 from the PPA. Gnome platform 3.22.

**Comment by alexlarsson** [comment]
I think its inheriting some of your bash properties, such as PS1 or something which depends on some tool i /usr on the host.

**Comment by matzipan** [comment]
If I do this with Builder, my prompt is `matzipan@matzipan-thinkpad:~$`. 
If I do this with Epiphany, my prompt is `bash-4.3$`.

But I get the same error in both, so something tells me it's not about my profile. Unless there's something which I'm misunderstanding about how bash works.

**Comment by alexlarsson** [comment]
I don't think its PS1 exactly, but something similar. Seems to be related to:
https://raw.githubusercontent.com/rcaloras/bash-preexec/master/bash-preexec.sh

**Comment by matzipan** [comment]
From what I understand, those bash functions are defined and the shell invokes them on precmd and preexec. 

Right. In my distro (elementary OS), there is a /etc/bash.bashrc file which then executes /etc/bash_preexec which is where those get defined.

**Comment by alexlarsson** [comment]
Yeah, but in the app sandbox /etc is from the runtime, so they will not be found.

**Comment by matzipan** [comment]
Then my guess is that this is a Gnome runtime issue?

**Comment by alexlarsson** [comment]
Its not really an "issue". Your host environment is referencing an executable that doesn't exist in the sandbox. This could happen with any file, like if you set PROMPT_COMMAND=/usr/bin/my-special-command then you can't expect that to be in the sandbox.

**Comment by matzipan** [comment]
True, but it seems that bash is depending on having these defined.  Which is why I'm thinking it's an issue with the Gnome runtime.

**Comment by Himura2la** [comment]
@alexlarsson, Hi, I surprisingly found myself stuck with the similar issue. I tried to use the terminal windows in Visual Studio Code and found that I can't access my software. 
```
make: avr-gcc: Command not found
```
I know it's realted to the sandbox, but I don't need these restrictions, how to access my real file system??

**Comment by Zhaojiajie1** [comment]
I have the same questioin ...... how to solve it  QAQ

**Comment by nbenitez** [comment]
> This could happen with any file, like if you set PROMPT_COMMAND=/usr/bin/my-special-command then you can't expect that to be in the sandbox.

I have exactly this problem. Solution is instead of calling `flatpak run --command=sh --devel your-app-id` do it like  `flatpak run --unset-env=PROMPT_COMMAND --command=sh --devel your-app-id`

But it would be nice if `flatpak run` could add extra parameters from a config file, so I don't have to add `--unset-env=PROMPT_COMMAND` every time I run `flatpak run` command. Specially since it's not possible to create a Bash alias for `flatpak run`.

**Comment by rusty-snake** [comment]
> Specially since it's not possible to create a Bash alias for flatpak run.

What are you talking about? It is possible. Just `alias flatpak-run="flatpak run --unset-env=PROMPT_COMMAND"`. I'm confused.

> But it would be nice if flatpak run could add extra parameters from a config file, so I don't have to add --unset-env=PROMPT_COMMAND every time I run flatpak run command. 

For me flatpak does not set `PROMPT_COMMAND` so I could not test, but did you tried `flatpak override --user --unset-env=PROMPT_COMMAND`? And it did not worked?
